src/main/java/io/vertx/ext/dropwizard/impl/HttpRequestMetric.java

```diff
@@ -12,10 +15,15 @@
   final String uri;
   HttpResponse response;
   long requestBegin;
+  final List<String> routes = new LinkedList<>();
```

Comment:
we should lazily create this list in case vertx web is not used to avoid instkantation of a list

Reply:
it should be of type Object and be a String if routing is called once, and List<String> if routed several times.

Reply:
we should also compute the sum of the routes so we can allocate a StringBuilder of the right size in getRoute() and use this builder to compute the String of getRoute to be more efficient and avoid un-necessary allocation

Reply:
Changed it accordingly. The resulting code isn't that easy to read though. Out of curiosity, this is a performance optimization and I wonder whether you requested those changes because a) you profiled or benchmarked the result or b) you are sure about the negative impact because of your experience with high-performance code and hot paths or c) of another reason I cannot see right now?

If you think it's worth it then we should do the same optimization for vertx-micrometer I guess? I prepared it already: https://github.com/vert-x3/vertx-micrometer-metrics/pull/122